grafana-plugin/src/components/TextEllipsisTooltip/TextEllipsisTooltip.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Tooltip } from '@grafana/ui';
+import cn from 'classnames/bind';
+
+import styles from 'assets/style/utils.css';
+
+const cx = cn.bind(styles);
+
+interface TextEllipsisTooltipProps {
+  content: string;
+  queryClassName?: string;
+  placement?: string;
+  className?: string;
+  children: JSX.Element | JSX.Element[];
+}
+
+const TextEllipsisTooltip: React.FC<TextEllipsisTooltipProps> = ({
+  queryClassName = 'overflow-child',
+  className,
+  content: textContent,
+  placement,
+  children,
+}) => {
+  const [isEllipsis, setIsEllipsis] = useState(true);
+  const elContentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    setEllipsis();
```

Comment:
it's called only onMount, but textContent and children can be changed later

Reply:
In current use cases the content cannot change, the only issue is on resizing as that will determine truncation, but that's sorted out by `onMouseHover`.

Reply:
maybe it's better to add [children] to useEffect deps right now?